src/main/java/uk/gov/hmcts/befta/util/MapVerifier.java

```diff
@@ -226,6 +231,19 @@ private Boolean canAcceptNullFor(Object expectedValue) {
         if (!(expectedValue instanceof String)) {
             return Boolean.FALSE;
         }
-        return !NOT_NULL.equalsIgnoreCase((String) expectedValue);
+        return nullable((String) expectedValue);
+    }
+
+    private boolean nullable(String strExpectedValue) {
+        ExpectedValuePlaceholder expectedValuePlaceholder = ExpectedValuePlaceholder.getByValue(strExpectedValue);
+        return expectedValuePlaceholder != null ?  expectedValuePlaceholder.isNullable(): true;
```

Comment:
pls avoid use of negation in logic checks.

Reply:
return expectedValuePlaceholder == null ?  true: expectedValuePlaceholder.isNullable();

This reads more smoothly with same effect.

Reply:
Ok